src/api/procedures/moveFunds.ts

```diff
@@ -35,22 +36,45 @@ export async function prepareMoveFunds(this: Procedure<Params, void>, args: Para
   } = this;
 
   const { from: fromPortfolio, to, items } = args;
-  const currentIdentity = await context.getCurrentIdentity();
 
-  let toPortfolio;
+  const {
+    owner: { did: fromDid },
+  } = fromPortfolio;
 
+  let toPortfolio;
   if (!to) {
-    toPortfolio = new DefaultPortfolio({ did: currentIdentity.did }, context);
+    toPortfolio = new DefaultPortfolio({ did: fromDid }, context);
   } else if (to instanceof BigNumber) {
-    toPortfolio = new NumberedPortfolio({ did: currentIdentity.did, id: to }, context);
+    toPortfolio = new NumberedPortfolio({ did: fromDid, id: to }, context);
   } else {
     toPortfolio = to;
   }
 
-  if (
-    fromPortfolio.owner.did === toPortfolio.owner.did &&
-    (fromPortfolio as NumberedPortfolio).id === (toPortfolio as NumberedPortfolio).id
-  ) {
+  const {
+    owner: { did: toDid },
+  } = toPortfolio;
+
+  const [fromPortfolioId, toPortfolioId, currentIdentity] = await Promise.all([
+    portfolioLikeToPortfolioId(fromPortfolio, context),
+    portfolioLikeToPortfolioId(toPortfolio, context),
+    context.getCurrentIdentity(),
+  ]);
+
+  if (fromDid !== toDid) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'Both portfolios should have the same owner',
+    });
+  }
+
+  if (currentIdentity.did !== fromDid) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'The current identity should be the owner of the origin portfolio',
+    });
+  }
+
+  if (isEqual(fromPortfolioId, toPortfolioId)) {
```

Comment:
Here you only need to check if the IDs are the same. You know the DIDs are equal because otherwise you would have thrown an error above